src/app/clusters/groups-server/groups-server.cpp

```diff
@@ -385,44 +423,36 @@ bool emberAfGroupsClusterRemoveGroupCallback(app::CommandHandler * commandObj, c
 bool emberAfGroupsClusterRemoveAllGroupsCallback(app::CommandHandler * commandObj, const app::ConcreteCommandPath & commandPath,
                                                  const Commands::RemoveAllGroups::DecodableType & commandData)
 {
-    EmberStatus sendStatus = EMBER_SUCCESS;
-    uint8_t i;
-    EndpointId endpoint = emberAfCurrentEndpoint();
-    bool success        = true;
+    FabricIndex fabricIndex    = GetFabricIndex(commandObj);
+    GroupDataProvider * groups = GetGroupDataProvider();
+    EndpointId endpointId      = commandPath.mEndpointId;
+    EmberStatus sendStatus     = EMBER_SUCCESS;
+    CHIP_ERROR err             = CHIP_ERROR_PERSISTED_STORAGE_VALUE_NOT_FOUND;
 
     emberAfGroupsClusterPrintln("RX: RemoveAllGroups");
 
-    for (i = 0; i < EMBER_BINDING_TABLE_SIZE; i++)
+    if (groups)
     {
-        EmberBindingTableEntry binding;
-        if (emberGetBinding(i, &binding) == EMBER_SUCCESS)
+#ifdef EMBER_AF_PLUGIN_SCENES
+        auto groupIter = groups->IterateGroupMappings(fabricIndex, endpointId);
+        if (nullptr != groupIter)
```

Comment:
Please de-nesdt this if with early error return if possible.

Reply:
I have de-nested the condition, but be aware there is a change in behavior: Now under faulty condition, no respond is sent. Nevertheless, this condition seems to be undefined under the spec.